### ISSUE TITLE
Fraction train param for val?

### ISSUE THREAD

**Issue by deneal123** [ultralytics/ultralytics#20481]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

There is a fraction parameter in the doc for the training pipe, but in the tuning format there is a forced validation, and its size is tens of times larger, a lot of time is lost. That is, if I have a 1M image split in the train and 0.2M in the validation, using fraction I can kill the size of the training, but not the validation. Is there a similar parameter for the validation sample?

**Comment by UltralyticsAssistant** [comment]
👋 Hello @deneal123, thank you for your interest in Ultralytics 🚀! We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://reddit.com/r/Ultralytics) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

---

🔔 This is an automated response. An Ultralytics engineer will review your question and assist you soon!

**Comment by deneal123** [comment]
> You're right - while the `fraction` parameter exists for training to reduce dataset size, there isn't a direct equivalent for validation. When tuning, validation is indeed required since metrics are needed to evaluate hyperparameter performance.
> 
> A workaround would be to create a smaller validation dataset specifically for tuning purposes. You could use the `split_classify_dataset()` function from the `ultralytics.data.split` module to create a smaller validation subset. Alternatively, you could modify your dataset's YAML file to point to a smaller validation set during tuning.
> 
> If you're working with very large datasets, you might also consider adjusting other validation parameters like `batch` size or using `rect=True` to speed up validation without reducing the dataset size. Would either of these approaches work for your situation?

Yes, thank you very much, the Split_classify_dataset() function is quite suitable, if with millions of images everything is still bearable, then in the case of a large number it becomes extremely inconvenient to duplicate new data in a new directory :), when the space on the hard drive is very tight, it would be very convenient with a similar parameter for training.

**Comment by glenn-jocher** [comment]
I understand your concern about disk space with large datasets. Instead of creating duplicate files, you could also:

1. Create symbolic links rather than copies using your own script (avoids duplicating data)
2. Temporarily modify your dataset YAML file during tuning to point to a subset of validation images

A `val_fraction` parameter would indeed be useful. If you'd like to see this feature added, consider opening a feature request in the repository. Would either of these approaches help with your storage constraints?
